### PR TITLE
chore: Document workaround for Raspberry Pi 3 dunfel to kirkstone update

### PR DESCRIPTION
**Pull request by lluiscampos**
Ticket: MEN-5740

### CODE REVIEW

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 1]
```diff
@@ -321,9 +321,41 @@ The block below shows 3 example artifacts.
...
+### Deploying an update from `dunfell` to `kirkstone` on Raspberry Pi blocks forever with `Waiting for root device /dev/mmcblk0p3...`
+
+Raspberry Pi boards have a set of boot firmware files (including Device Tree Blob files) that are
+located on the vfat boot part, and these files are not updated when you perform an update of the
+root filesystem.
+
+Occasionally there will be changes to the Raspberry Pi software stack that requires that these files
+are update. Typically there is something in the Linux kernel that requires something that is in the
+boot firmware. This means that to update the Linux kernel you must also update the boot firmware
+files.
```
Comment (tranchitella) [inline 301.Troubleshoot/03.Mender-Client/docs.md:341]:
```suggestion
Occasionally, changes to the Raspberry Pi software stack require that these files are
updated. Typically there are dependencies in the Linux kernel on something in the boot
firmware. Therefore, to update the Linux kernel, you must also update the boot firmware
files.
```

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 2]
```diff
@@ -321,9 +321,41 @@ The block below shows 3 example artifacts.
...
+
+<!--AUTOVERSION: "between `%` and `%` yocto"/ignore -->
+This is what has happened between `dunfell` and `kirkstone` yocto releases for Raspberry Pi 3B
+(other boards might have been affected too).
+
+Mender provides a recipe to include an state script in the Artifact that will copy the firmware
```
Comment (tranchitella) [inline 301.Troubleshoot/03.Mender-Client/docs.md:347]:
```suggestion
Mender provides a recipe to include a state script in the Artifact that will copy the firmware
```

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 3]
```diff
@@ -321,9 +321,41 @@ The block below shows 3 example artifacts.
...
+files into the boot during the Artifact install. Enable it by adding the following to e.g
+`local.conf`:
+
+```
+INHERIT += "rpi-update-firmware"
+```
+
+The script can be overridden to customize what
+files to update, e.g only DTB files. See [mendersoftware/meta-mender@5abe2e7](https://github.com/mendersoftware/meta-mender/commit/5abe2e7b1f85b3482cb53b747ae9ced3f20a3e73) for more details.
```
Comment (tranchitella) [inline 301.Troubleshoot/03.Mender-Client/docs.md:356]:
```suggestion
You can override the script customizing what files to update, e.g only DTB files.
See [mendersoftware/meta-mender@5abe2e7](https://github.com/mendersoftware/meta-mender/commit/5abe2e7b1f85b3482cb53b747ae9ced3f20a3e73) for more details.
```

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 4]
```diff
@@ -321,9 +321,41 @@ The block below shows 3 example artifacts.
...
+
+!!! Updating the boot firmware files can not be done atomically and the files are not roll-backed
+!!! even though an rootfs rollback is performed. Conclusion, this is an risky operating which could
+!!! brick your device. We recommend to use this workaround once, when upgrading between yocto
+!!! releases, and not for every update henceforth.
```
Comment (tranchitella) [inline 301.Troubleshoot/03.Mender-Client/docs.md:361]:
```suggestion
!!! Updating the boot firmware files can not be done atomically and the files are not rollbacked
!!! even though a rootfs rollback is performed. In conclusion, this is a risky operating which could
!!! brick your device. We recommend using this workaround once when upgrading between yocto
!!! releases, and not for every update henceforth.
```

**Review by tranchitella — COMMENTED** [review]
A few commends to improve the readability.

**Comment by lluiscampos** [comment]
Thank you @tranchitella for your comments - I fixed the text as you suggested :+1:

**Comment by lluiscampos** [comment]
I also added a new warning about how to clean the state script. We have it already [here](https://docs.mender.io/system-updates-yocto-project/image-customization/state-scripts#including-state-scripts-in-artifacts-and-disk-images) but just to be in the safe side.

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 5]
```diff
@@ -321,9 +321,46 @@ The block below shows 3 example artifacts.
...
 * artifact - `mender-artifact read <mender-artifact.mender>`
 * device - Run the command on the device `mender show-provides`
 * server UI - `Releases -> Select Release -> Expand the artifact info by clicking it -> Expand Provides and Depends`
+
+
+<!--AUTOVERSION: "from `%` to `%` on"/ignore -->
+### Deploying an update from `dunfell` to `kirkstone` on Raspberry Pi blocks forever with `Waiting for root device /dev/mmcblk0p3...`
+
+Raspberry Pi boards have a set of boot firmware files (including Device Tree Blob files) that are
+located on the vfat boot part, and these files are not updated when you perform an update of the
```
Comment (TheYoctoJester) [inline 301.Troubleshoot/03.Mender-Client/docs.md:335]: "boot partition" please, "boot part" is jargon.

Comment (oleorhagen) [inline 301.Troubleshoot/03.Mender-Client/docs.md:335]: Q: Why vfat explicitly?

Comment (TheYoctoJester) [inline 301.Troubleshoot/03.Mender-Client/docs.md:335]: "boot partition" please, "boot part" is jargon

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 6]
```diff
@@ -321,9 +321,46 @@ The block below shows 3 example artifacts.
...
+root filesystem.
```
Comment (oleorhagen) [inline 301.Troubleshoot/03.Mender-Client/docs.md:336]: S: Add something like: `where the kernel lives`.

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 7]
```diff
@@ -321,9 +321,46 @@ The block below shows 3 example artifacts.
...
+
+```
+INHERIT += "rpi-update-firmware"
+```
+
+You can override the script customizing what files to update, e.g only DTB files. See
+[mendersoftware/meta-mender@5abe2e7](https://github.com/mendersoftware/meta-mender/commit/5abe2e7b1f85b3482cb53b747ae9ced3f20a3e73)
+for more details.
+
+!! Updating the boot firmware files can not be done atomically and the files are not rollbacked
```
Comment (oleorhagen) [inline 301.Troubleshoot/03.Mender-Client/docs.md:358]: S: `rolled back`

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 8]
```diff
@@ -321,9 +321,46 @@ The block below shows 3 example artifacts.
...
+!! even though a rootfs rollback is performed. In conclusion, this is a risky operating which could
```
Comment (oleorhagen) [inline 301.Troubleshoot/03.Mender-Client/docs.md:359]: `s/operating/operation`

**Review by oleorhagen — APPROVED** [review]

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 9]
```diff
@@ -321,9 +321,46 @@ The block below shows 3 example artifacts.
...
+
+<!--AUTOVERSION: "from `%` to `%` on"/ignore -->
+### Deploying an update from `dunfell` to `kirkstone` on Raspberry Pi blocks forever with `Waiting for root device /dev/mmcblk0p3...`
+
+Raspberry Pi boards have a set of boot firmware files (including Device Tree Blob files) that are
+located on the vfat boot part, and these files are not updated when you perform an update of the
+root filesystem.
+
+Occasionally, changes to the Raspberry Pi software stack require that these files are updated.
+Typically there are dependencies in the Linux kernel on something in the boot firmware. Therefore,
```
Comment (TheYoctoJester) [inline 301.Troubleshoot/03.Mender-Client/docs.md:339]: "on something" sounds quite handwavy. Maybe better "Between the Linux kernel and the boot firmware, version interdependencies exist which need to be taken care of."

File: 301.Troubleshoot/03.Mender-Client/docs.md (hunk 7 above)
Comment (TheYoctoJester) [inline 301.Troubleshoot/03.Mender-Client/docs.md:358]: "are not rollbacked" -> "cannot be rolled back even if a rootfs rollback might be performed"

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 10]
```diff
@@ -321,9 +321,46 @@ The block below shows 3 example artifacts.
...
+INHERIT += "rpi-update-firmware"
+```
+
+You can override the script customizing what files to update, e.g only DTB files. See
+[mendersoftware/meta-mender@5abe2e7](https://github.com/mendersoftware/meta-mender/commit/5abe2e7b1f85b3482cb53b747ae9ced3f20a3e73)
+for more details.
+
+!! Updating the boot firmware files can not be done atomically and the files are not rollbacked
+!! even though a rootfs rollback is performed. In conclusion, this is a risky operating which could
+!! brick your device. We recommend using this workaround once when upgrading between Yocto
```
Comment (TheYoctoJester) [inline 301.Troubleshoot/03.Mender-Client/docs.md:360]: "only once"

**Review by TheYoctoJester — COMMENTED** [review]
Just some minor wording things, all in all I like the explanation very much. If you disagree with the suggestions, I will also accept the original.

File: 301.Troubleshoot/03.Mender-Client/docs.md (hunk 5 above)
Comment (lluiscampos) [reply 301.Troubleshoot/03.Mender-Client/docs.md:335]: I copied it from the commit message. I'll remove it.

**Pull request merged**